apps/bare-expo/android/app/build.gradle

```diff
@@ -76,9 +76,12 @@ import com.android.build.OutputFile
 
 def projectRoot = rootDir.getAbsoluteFile().getParentFile().getAbsolutePath()
 
+def androidSchemes = gradle.ext.expoGetConfig('android.scheme', [ "json": true ])
+println "android schemes: ${androidSchemes}"
+
 project.ext.react = [
     entryFile: ["node", "-e", "require('expo/scripts/resolveAppEntry')", projectRoot, "android"].execute(null, rootDir).text.trim(),
-    enableHermes: (findProperty('expo.jsEngine') ?: "jsc") == "hermes",  // clean and rebuild if changing
+    enableHermes: (gradle.ext.expoGetConfig("android.jsEngine") ?: gradle.ext.expoGetConfig("jsEngine") ?: "jsc") == "hermes",  // clean and rebuild if changing
```

Comment:
## Issue 1

One notable issue here is that we are only reading the synchronous config (i.e. no mods are being evaluated). This essentially means anything that was writing to `mods.android.gradleProperties` (`withGradleProperties`) is now being ignored.

We moved to mods for prebuild to avoid mutation issues where running `expo prebuild` would persist all static changes into the local `app.json`.

Example:

`app.json`
```json
{
  "expo": {
    "plugins": ["./custom"]
  }
}
```

`custom.js`
```js
const withFooBar = (config) => {
  config.foobar = true;
  return config;
}
```

- Runs `expo prebuild`
- `app.json` may be persisted with `foobar` locally, this happens when we write the `ios.bundleIdentifier` automatically, etc.

`app.json`
```json
{
  "expo": {
    "foobar": true,
    "plugins": ["./custom"]
  }
}
```